OIDC Protocol Mappers with same claim

cgeorgilakis opened the pull request:
Closes #25774

<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

Commit:
OIDC Protocol Mappers with same claim

Closes #25774

Signed-off-by: cgeorgilakis-grnet <cgeorgilakis@admin.grnet.gr>

rmartinc commented on services/src/main/java/org/keycloak/protocol/oidc/mappers/OIDCAttributeMapperHelper.java:
```diff
@@ -309,8 +311,24 @@ private static void mapClaim(List<String> split, Object attributeValue, Map<Stri
         for (String component : split) {
             i++;
-            if (i == length) {
+            if (i == length && length == 1 && primaryClaims.contains(split.get(0))) {
```
Instead of using the hard-coded `primaryClaims` list maybe we can better rely on the multi-valued configuration option. Those primary claims will never be defined as multi (and if they are they are probably wrong defined). You can pass the model to the method because it's private without problems.

@cgeorgilakis WDYT? Do you see any problem with that approach?

rmartinc reviewed (changes requested):
Thanks @cgeorgilakis for the PR!

I have just added one comment because I don't like the hard-coded `primaryClaims` list. My proposal is using the `isMultivalued` from the config option. Let me know if you see any problem with that idea.

cgeorgilakis commented:
> Thanks @cgeorgilakis for the PR!
> 
> I have just added one comment because I don't like the hard-coded `primaryClaims` list. My proposal is using the `isMultivalued` from the config option. Let me know if you see any problem with that idea.

I agree and I have changed it. Tests are ok.

rmartinc commented on services/src/main/java/org/keycloak/protocol/oidc/mappers/OIDCAttributeMapperHelper.java:
```diff
@@ -307,16 +308,32 @@ private static <T> void mapClaim(T token, ProtocolMapperModel mappingModel, Obje
         for (String component : split) {
             i++;
-            if (i == length) {
+            if (i == length && length == 1 && !isMultivalued) {
```
I don't understand the `length == 1` here, is it needed? The attribute should be appended if multi-valued no matter if the claim is `claim` or `complex.claim` (length of the split), am I right?

cgeorgilakis replied:
You are right. I have been removed  it.

rmartinc reviewed (changes requested):
Thanks @cgeorgilakis! I have just one more doubt about the PR...

rmartinc reviewed (approved):
Thanks @cgeorgilakis! LGTM now...
